src/main/groovy/com/github/jrubygradle/internal/GemVersionResolver.groovy

```diff
@@ -4,10 +4,52 @@ import org.gradle.api.artifacts.DependencyResolveDetails
 
 class GemVersionResolver {
 
-    private Map versions = [:]
-  
+    Map versions
+    private final Object configuration
+    private final Object logger
+
+    static void setup(Object project) {
+        project.configurations.each {
```

Comment:
thinking about this a bit: it adds a resolver for each configuration but the resolver does touch only artifacts with groupId == 'rubygems'. on any ```compile```, ```test``` or whatever there is the resolution does not change. actually even for rubygems it does not change the dependency tree since that is gradle does, it only ensures if that gradle picks the newest version which fits to all version ranges of the same artifact found at several places inside the tree.

but again on rubygems:* artifacts are touched. IMO it is OK as it is.